File: backend/src/main/java/com/spillabs/sales_order/service/interfaces/ItemService.java
package com.spillabs.sales_order.service.interfaces;

import com.spillabs.sales_order.dto.ItemDTO;
import java.util.List;

public interface ItemService {
    List<ItemDTO> getAllItems();

    ItemDTO createItem(ItemDTO itemDTO);

    ItemDTO getItemById(Long id);

    ItemDTO getItemByCode(String code);
}

File: frontend/src/services/itemService.js
import api from './api';

export const itemService = {
    getAllItems: async () => {
        const response = await api.get('/api/items');
        return response.data;
    },

    getItemById: async (id) => {
        const response = await api.get(`/api/items/${id}`);
        return response.data;
    },

    createItem: async (itemData) => {
        const response = await api.post('/api/items', itemData);
        return response.data;
    },
};
